pkg/api/validation/validation.go

```diff
@@ -934,6 +937,21 @@ func validateProjectedVolumeSource(projection *api.ProjectedVolumeSource, fldPat
 	return allErrs
 }
 
+func validateHostPathType(hostPathType api.HostPathType, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	switch hostPathType {
+	case api.HostPathUnset, api.HostPathExists, api.HostPathFile, api.HostPathDevice, api.HostPathSocket, api.HostPathDirectory:
+		break
+	case api.HostPathNewDir, api.HostPathCharDev, api.HostPathBlockDev, api.HostPathNewFile, api.HostPathOptional:
+		allErrs = append(allErrs, field.Invalid(fldPath, hostPathType, "is excluded"))
```

Comment:
Why are these excluded?  everything you define is supported.  all you need to check is that the value is in the set of values you support.  typically you do this with an out-of-function `var supportedHostPathTypes = sets.String(A, B, C, ...)` and then test `supportedHostPathTypes.Has(hptype)`